### PR TITLE
Multiarch installation fix

### PR DESCRIPTION
**Pull request by e-fominov**
Earlier we have added support for GNUInstallDirs, but haven't changed the CMakeLists.txt to produce correct paths in dlibConfig.cmake

When trying to optimize NEON code, I have found a problem using installed dlib on Tegra TK1:
cmake_error

To reproduce the error you need to use dlib installed by CMake (cmake --build . --target install) in your project on some multiarch-enabled machine

On Tegra library is install into /usr/local/lib/arm-linux-gnueabihf, while on normal Linux machines we have this install dir: /usr/local/lib,

Dlib is using relative paths when configuring dlibConfig.cmake file:
```CMake
set(CONF_INSTALL_PATH "\${dlib_CMAKE_DIR}/../../../")
```

This PR makes dlib to correctly install on Tegra TK 1 machine and keeps it work on normal Linuxes

### CODE REVIEW

File: dlib/cmake_utils/dlibConfig.cmake.in [hunk 1]
```diff
@@ -23,20 +23,20 @@
 get_filename_component(dlib_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
 
 if(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} VERSION_LESS 2.8)
-    get_filename_component(dlib_INSTALL_PATH "@CONF_INSTALL_PATH@" ABSOLUTE)
+    get_filename_component(dlib_INSTALL_PATH "@CMAKE_INSTALL_PREFIX@" ABSOLUTE)
```
Comment (davisking) [inline dlib/cmake_utils/dlibConfig.cmake.in]: Do we still need these get_filename_component lines at all if we switch to this method?

**Comment by e-fominov** [comment]
I was trying to make minimal changes. May be we can make this file more simple

**Comment by davisking** [comment]
Cool.  I just cleaned it up and it seems to be fine on Ubuntu.  Can you make this PR editable to me so I can push the changes to it?  Then you can make sure they aren't messing up the builds you are doing and we can merge it to the main repo.

**Comment by davisking** [comment]
Oops.  I pushed to your main branch.  Guess I should have pushed to the branch for this PR.  The changes are there though.

**Comment by davisking** [comment]
Also, there is this: https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/ which is pretty convenient.

**Comment by e-fominov** [comment]
This option is already enabled

**Comment by e-fominov** [comment]
PR now contains your changes from master branch

**Comment by davisking** [comment]
Cool.  Let me know when you think it's ready to merge.

**Comment by e-fominov** [comment]
Successfully installed on Ubuntu 16 and Ubuntu 14 (Tegra TK1/ARM) with latest changes. I think this PR can be merged

**Comment by davisking** [comment]
Sweet.  Thanks for the PR :)

**Pull request merged**